buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -714,20 +714,94 @@ public static ByteBuf readBytes(ByteBufAllocator alloc, ByteBuf buffer, int leng
         }
     }
 
-    static int lastIndexOf(AbstractByteBuf buffer, int fromIndex, int toIndex, byte value) {
+    static int lastIndexOf(final AbstractByteBuf buffer, int fromIndex, final int toIndex, final byte value) {
         assert fromIndex > toIndex;
         final int capacity = buffer.capacity();
         fromIndex = Math.min(fromIndex, capacity);
-        if (fromIndex < 0 || capacity == 0) {
+        if (fromIndex <= 0) { // fromIndex is the exclusive upper bound.
             return -1;
         }
-        buffer.checkIndex(toIndex, fromIndex - toIndex);
+        final int length = fromIndex - toIndex;
+        buffer.checkIndex(toIndex, length);
+        if (!PlatformDependent.isUnaligned()) {
+            return linearLastIndexOf(buffer, fromIndex, toIndex, value);
+        }
+        final int longCount = length >>> 3;
+        int offset = fromIndex;
+        if (longCount > 0) {
+            final ByteOrder nativeOrder = ByteOrder.nativeOrder();
+            final boolean isNative = nativeOrder == buffer.order();
+            final boolean useLE = nativeOrder == ByteOrder.LITTLE_ENDIAN;
+            final long pattern = SWARUtil.compilePattern(value);
+            for (int i = 0; i < longCount; i++) {
+                // use the faster available getLong
+                final long word = useLE? buffer._getLongLE(offset - Long.BYTES)
+                        : buffer._getLong(offset - Long.BYTES);
```

Comment:
While searching backward, we need to check the last occurrence of the needle in the long batch, which means basically working with the opposite endianness. I don't see any mention about it (in a comment too)

Reply:
I keep the endianness here but reverted when I extract index from the result (`return offset - 1 - SWARUtil.getIndex(result, !isNative)`).
Let me add a comment there.

Reply:
Done. PTAL :D

Reply:
@franz1981 
Done. PTAL :D
```
                if (result != 0) {
                    // used the oppoiste endianness since we are looking for the last index.
                    return offset - 1 - SWARUtil.getIndex(result, !isNative);
                }
```